external-crates/move/move-compiler/src/typing/translate.rs

```diff
@@ -2265,3 +2267,161 @@ fn make_arg_types<S: std::fmt::Display, F: Fn() -> S>(
     }
     given
 }
+
+//**************************************************************************************************
+// Module-wide warnings
+//**************************************************************************************************
+
+/// Generates warnings for unused struct types and unused (private) functions.
+fn gen_unused_warnings(context: &mut Context, ident: ModuleIdent, mdef: &T::ModuleDefinition) {
+    let mut packed_types = BTreeSet::new();
+    let mut called_fns = BTreeSet::new();
+
+    for (_, _, fun) in &mdef.functions {
+        let sp!(_, body) = &fun.body;
+        if let T::FunctionBody_::Defined(seq) = body {
+            for item in seq {
+                let sp!(_, seq_item) = item;
+                seq_item_collect_used(&seq_item, &ident, &mut called_fns, &mut packed_types)
+            }
+        }
+    }
+
+    context
+        .env
+        .add_warning_filter_scope(mdef.warning_filter.clone());
+
+    for (loc, name, fun) in &mdef.functions {
+        context
+            .env
+            .add_warning_filter_scope(fun.warning_filter.clone());
+        if !called_fns.contains(name) {
+            // TODO: postponing handling of friend functions until we decide what to do with them
+            // vis-a-vis ideas around package-private
+            if let Visibility::Internal = fun.visibility {
+                if fun.entry.is_none() {
```

Comment:
nit: you can probably do this all in one if with `if !called_fns.contains(name) && matches!(fun.visibility, Visibility::Internal) && fun.entry.is_none()`